internal/provider/resource_vault_cluster.go

```diff
@@ -120,14 +126,228 @@ func resourceVaultCluster() *schema.Resource {
 	}
 }
 
-func resourceVaultClusterCreate(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+func resourceVaultClusterCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*clients.Client)
+
+	clusterID := d.Get("cluster_id").(string)
+	hvnID := d.Get("hvn_id").(string)
+	loc := &sharedmodels.HashicorpCloudLocationLocation{
+		OrganizationID: client.Config.OrganizationID,
+		ProjectID:      client.Config.ProjectID,
+	}
+
+	// Use the hvn to get provider and region.
+	hvn, err := clients.GetHvnByID(ctx, client, loc, hvnID)
+	if err != nil {
+		return diag.Errorf("unable to find existing HVN (%s): %v", hvnID, err)
+	}
+	loc.Region = &sharedmodels.HashicorpCloudLocationRegion{
+		Provider: hvn.Location.Region.Provider,
+		Region:   hvn.Location.Region.Region,
+	}
+
+	// Check for an existing Vault cluster.
+	_, err = clients.GetVaultClusterByID(ctx, client, loc, clusterID)
+	if err != nil {
+		if !clients.IsResponseCodeNotFound(err) {
+			return diag.Errorf("unable to check for presence of an existing Vault cluster (%s): %v", clusterID, err)
+		}
+
+		// A 404 indicates a Vault cluster was not found.
+		log.Printf("[INFO] Vault cluster (%s) not found, proceeding with create", clusterID)
+	} else {
+		return diag.Errorf("a Vault cluster with cluster_id=%q in project_id=%q already exists - to be managed via Terraform this resource needs to be imported into the State.  Please see the resource documentation for hcp_vault_cluster for more information.", clusterID, loc.ProjectID)
+	}
+
+	// TODO: Recommended version is hard-coded for now, but eventually should be fetched from an API.
+	vaultVersion := "1.7.0"
```

Comment:
Instead of hardcoding here, what do you think of just leaving `InitialVersion` out of the request if `min_vault_version` isn't supplied, and letting the HCP Vault backend handle this? In this case, it would create a cluster with the default version.

Reply:
Oh nice, didn't know it set a default. That sounds good to me